service/awsconfig/v4/cloudconfig/cloud_config.go

```diff
@@ -39,9 +44,21 @@ func New(config Config) (*CloudConfig, error) {
 		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
 	}
 
+	var k8sAPIExtraArgs []string
+	{
+		for _, arg := range config.K8sAPIExtraArgs {
+			if !strings.HasSuffix(arg, "=") {
+				k8sAPIExtraArgs = append(k8sAPIExtraArgs, arg)
+			} else {
+				config.Logger.Log("warning", "api-server arguments must not be empty", "argument", arg)
+			}
+		}
+	}
```

Comment:
I wouldn't do that and add check below instead...

Reply:
I did it in this way, because I wanted to bound change amap to cloudconfig resource, as this change is related only to parameters, which should eventually impact k8scloudconfig only. Another option is to move it to the `NewFramework` function, where we validate most config parameters.